### PR TITLE
ci: include multiple versions of java and OS in test_pack_dock workflow

### PR DESCRIPTION
**Pull request by lionel-nj**
**Summary:**

This PR provides support to execute `test_pack_doc.yml` workflow on multiple versions of Java: 11 and 17, as well as different OS; in order to flag potential incompatibilities. 

**Expected behavior:** 

Run the same jobs `test`, `doc`, and `pack` on different versions of the JDK: 11 and 17.

All artifacts should be persisted and identified by the JDK's and OS version that was used to produce them.

Capture d’écran, le 2022-02-01 à 12 53 14

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by lionel-nj** [comment]
@ed-g I'd like to add to add the OS you're using to the list of OS to be run. You mentioned Linux x86 [earlier](https://github.com/MobilityData/gtfs-validator/pull/1086#issuecomment-1022506490), could you give information about the distribution that is used on your side please?

- [List of OS supported by Github Actions](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources)

File: .github/workflows/test_pack_doc.yml
Comment (barbeau) [reply .github/workflows/test_pack_doc.yml:23]: Given that we're using `ubuntu-latest`, you could use `windows-latest` instead.

**Comment by ed-g** [comment]
> @ed-g I'd like to add to add the OS you're using to the list of OS to be run. You mentioned Linux x86 [earlier](https://github.com/MobilityData/gtfs-validator/pull/1086#issuecomment-1022506490), could you give information about the distribution that is used on your side please?

You bet, it's centos stream 8 (redhat workalike) here's the version from the server;

uname
`Linux v 4.18.0-358.el8.x86_64 #1 SMP x86_64 x86_64 GNU/Linux`

redhat-release
`CentOS Stream release 8`

java packages
```
java-17-openjdk-headless-17.0.1.0.12-2.el8_5.x86_64
java-17-openjdk-17.0.1.0.12-2.el8_5.x86_64
java-17-openjdk-devel-17.0.1.0.12-2.el8_5.x86_64
```

**Comment by ed-g** [comment]
"Ubuntu 20.04" is probably the nearest.

**Comment by lionel-nj** [comment]
It seems that the use of emoji is problematic on `window-latest` when it comes to generating the javadoc. I opened https://github.com/MobilityData/gtfs-validator/issues/1101 to document this issue for the future work. As of now `doc` job is only run on `ubuntu-latest` and `ubuntu-20.04`

**Comment by lionel-nj** [comment]
It seems that the generation of javadocs on `window-latest` is problematic due to use of emojis in `String`s. I opened #1101  to document this for future work. 

As of now, `doc` job of `test_pack_doc` workflow is only executed on `ubutu-latest` and `ubuntu-20.04`. After this PR is merge we will potentially be able to replicate the bug regarding unit tests in `output-comparator` from https://github.com/MobilityData/gtfs-validator/pull/1086#issuecomment-1026223882.

**Comment by barbeau** [comment]
Hopefully Windows Javadoc generation should work correctly after PR https://github.com/MobilityData/gtfs-validator/pull/1121 is merged, which should unblock this PR.

**Review by barbeau — APPROVED** [review]
@maximearmstrong Do you have any objection to merging this? It seems to be working as intended to me - package/test/javadoc tasks are run across Linux and Windows on GitHub CI now, for both Java 11 and 17.

**Comment by maximearmstrong** [comment]
@barbeau All good on my side! Thank you!

**Pull request merged**
